### PR TITLE
fix: add various small fixes to the code

### PR DESCRIPTION
**Pull request by RichardBray**
# Description

The purpose of this PR is to fix a few small typos and address an issue with a file limit error.

I've tried my best to make the code as readable as possible but if any of it doesn't make sense please don't hesitate to ask.

### CODE REVIEW

File: api/controllers/FileUploadController.js [hunk 1]
```diff
@@ -49,8 +49,9 @@ const FileUploadController = {
             const userData = HelperService.getUserData(req, res);
             const displayFilenameErrors = req.flash('displayFilenameErrors');
             const infectedFiles = req.flash('infectedFiles');
-            let genericErrors = req.flash('genericErrors');
+            let genericErrors = req.flash('genericErrors') ?? [];
```
Comment (jenbutongit) [inline api/controllers/FileUploadController.js:54]: 💪 `??`

File: api/controllers/FileUploadController.js [hunk 2]
```diff
@@ -61,7 +62,11 @@ const FileUploadController = {
                 return res.forbidden();
             }
 
-            FileUploadController._maxFileLimitCheck(req);
+
+            if (!maxFileLimitErrorExists) {
+                FileUploadController._maxFileLimitCheck(req);
+                genericErrors = req.flash('genericErrors');
```
Comment (jenbutongit) [inline api/controllers/FileUploadController.js]: what happens when there are genericErrors, but you don't have max file limit errors? won't this end up replacing them?

File: api/controllers/FileUploadController.js [hunk 3]
```diff
@@ -49,8 +49,9 @@ const FileUploadController = {
             const userData = HelperService.getUserData(req, res);
             const displayFilenameErrors = req.flash('displayFilenameErrors');
             const infectedFiles = req.flash('infectedFiles');
-            let genericErrors = req.flash('genericErrors');
+            let genericErrors = req.flash('genericErrors') ?? [];
             let backLink = '/completing-your-application';
+            const maxFileLimitErrorExists = genericErrors.includes(POST_UPLOAD_ERROR_MESSAGES.fileCountError);
```
Comment (jenbutongit) [inline api/controllers/FileUploadController.js]: you have a lot of logic finding/dealing with a specific error. should fileCountError have it's own req.flash key?

**Review by jenbutongit — CHANGES_REQUESTED** [review]

File: api/controllers/FileUploadController.js (hunk 2 above)
Comment (RichardBray) [reply api/controllers/FileUploadController.js]: ah that's a fair point, I'll address that

File: api/controllers/FileUploadController.js (hunk 3 above)
Comment (RichardBray) [reply api/controllers/FileUploadController.js]: yep I can do that, might fix the issue that you raised in the previous comment as well

**Review by jenbutongit — APPROVED** [review]

**Pull request merged**